### ISSUE TITLE
Admin Center support for config validation

### ISSUE THREAD

**Issue by k8vance88** [OpenLiberty/open-liberty#18696]
## Description of the high level feature, including any external spec links:  
Support for config validators in Admin Center server config view. A test connection button will be displayed for resources that support config validation. The UI will require resources to be saved to the config before allowing the user to click the test connection button. 

The pre-req REST feature was done here: https://github.com/OpenLiberty/open-liberty/issues/6509 and https://bigblue.aha.io/features/APSFOUND-22

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
## When complete & mandatory, add links to the UFO (Upcoming Feature Overview) document, FTS (Feature Test Summary), blogs post issues(s), and [Aha](https://cloud-platform.ideas.aha.io/) (externally raised RFEs):
- UFO: https://ibm.box.com/s/0f2w5ujd3oeb6vzfjnezdij77wb0gews
- FTS: https://github.com/OpenLiberty/open-liberty/issues/23807
- Beta Blog Post (if applicable): https://github.com/OpenLiberty/open-liberty/issues/23629
- Blog Post: https://github.com/OpenLiberty/open-liberty/issues/24124
- Aha: https://bigblue.aha.io/features/APSFOUND-22
   - Also, add a comment into the Aha idea linking to this issue.

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
## List of Steps to complete or get approvals / sign-offs for Onboarding to the Liberty release (GM date)

Instructions:
- Do the actions below and mark them complete in the checklist when they are done.
- Make sure all feature readiness approvers put the appropriate tag on the epic to indicate their approval.

## **Design**
Before Development Starts or 8 weeks before Onboarding
- [x] POC Design / UFO Review Scheduled (David Chang) or N/A.
- [x] POC Design / UFO Reviewed (Feature Owner) or N/A.
- [x] Complete any follow-ons from the POC Review.
- [x] Design / UFO Approval ([Alasdair Nottingham](https://github.com/NottyCode)) or N/A.
- [x] No Design / No UFO Approval ([Alasdair Nottingham](https://github.com/NottyCode)) or N/A.
- [x] SVT Requirements identified. (Epic owner / Feature owner with SVT focal point)
- [x] ID Requirements identified ([Documenting Open Liberty](https://github.com/OpenLiberty/open-liberty/wiki/Documenting-Open-Liberty)). (Epic owner / Feature owner with ID focal point)
- [x] Create a child task of this epic entitled "Feature Test Summary" via [this template](https://github.com/OpenLiberty/open-liberty/issues/new?assignees=&labels=Feature+Test+Summary&template=feature_test_summary.md&title=). Add the link in above.

##
##

## **Development**
When active development has begun 
- [x] Add the "In Progress" label to this issue.

## **Beta**
If your feature, or portions of it, are going to be included in a beta  
Before Onboarding the beta
- [x] Beta Fence the functionality (`kind=beta`, `ibm:beta`, `ProductInfo.getBetaEdition()`)  

1 week before beta GA
- [x] Create, populate, and link to the [Beta blog post issue](https://github.com/OpenLiberty/open-liberty/issues/new?assignees=lauracowen%2C+jakub-pomykala&labels=&template=blog_post_beta.md&title=BETA+BLOG+-+title_of_your_update)

## **Legal**
3 weeks before Onboarding
- [x] Identify all open source libraries that are changing or are new. Work with Legal Release Services (Cass Tucker or Release PM) to get open source cleared and approved. Or N/A. (Epic Owner).   New or changed open source impacts license and Certificate of Originality.

## **Translation**
3 weeks before Onboarding
- [x] All new or changed PII messages are checked into the integration branch, before the last translation shipment out. (Epic Owner)

## **Feature Complete**
2 weeks before Onboarding
- [x] Implementation complete. (Epic owner / Feature owner)
- [x] All function tests complete. Ready for FAT Approval. (Epic owner / Feature owner)
- [x] Review all known issues for Stop Ship. (Epic owner / Feature owner / PM)

## **Focal Point Approvals**
2 to 1 week before Onboarding
#### You **MUST** have the Design Approved or No Design Approved label before requesting focal point approvals.

All features (both "Design Approved" and "No Design Approved")
- [x] **FAT** - ([OpenLiberty/fat-approvers](https://github.com/orgs/OpenLiberty/teams/fat-approvers)). SOE FATS are running successfully or N/A . Approver adds label focalApproved:fat to the Epic in Github.
- [x] **Demo** - ([Tom Evans](https://github.com/tevans78) or [Chuck Bridgham](https://github.com/cbridgha)). Demo is scheduled for an upcoming EOI. Approver adds label focalApproved:demo to the Epic in Github.
- [x] **Globalization** ([Sam Wong](https://github.com/samwatibm) - Liberty / Simy Cheeran - tWAS). Translation is complete or N/A. TVT - complete or N/A. Approver adds label focalApproved:globalization to the Epic in Github.

"Design Approved" features
- [x] **Accessibility** - ([Steven Zvonek](https://github.com/steven1046)). Accessibility testing is complete or N/A. Approver adds label focalApproved:accessibility to the Epic in Github.
- [x] **ID** - ([Karen Deen](https://github.com/chirp1)). Documentation work is complete or N/A . Approver adds label focalApproved:id to the Epic in Github.
- [x] **Performance** - ([Jared Anderson](https://github.com/jhanders34)). Performance testing is complete with no high severity defects or N/A . Approver adds label focalApproved:performance to the Epic in Github.
- [x] **Serviceability** - ([Don Bourne](https://github.com/donbourne)). Serviceability has been addressed.
- [x] **STE** - (Swati Kasundra). STE chart deck is complete or N/A . Approver adds label focalApproved:ste to the Epic in Github.
- [x] **SVT** - ([Brian Hanczaryk](https://github.com/hanczaryk) - APS). SVT is complete or N/A . Approver adds label focalApproved:svt to the Epic in Github.

## **Ready for GA**
1 week before Onboarding
- [x] No Stop Ship issues for the feature. (Epic owner / Feature owner / Release PM)
- [ ] Ship Readiness Review and Release Notes completed (Epic owner / Feature owner / Release PM)
- [x] Github Epic and Epic's issues are closed / complete. All PRs are committed to the release branch. (Epic owner / Feature owner / Backlog Subtribe PM)

1 week before GA
- [x] Create, populate, and link to the [Blog post issue](https://github.com/OpenLiberty/open-liberty/issues/new?assignees=lauracowen%2C+jakub-pomykala&labels=&template=blog_post_ga_release.md&title=GA+BLOG+-+title_of_your_update)

## **Other deliverbles**
- [x] **OL Guides** - ([Yee-Kang Chang](https://github.com/yeekangc)). Assessment for OL Guides is complete or N/A.
- [x] **WDT** - (Leonard Theivendra). WDT work complete or N/A.
- [ ] **Blog** - ([Laura Cowen](https://github.com/lauracowen)) Blog article writeup (Epic owner / Feature owner / Laura Cowen)

**Comment by k8vance88** [comment]
UFO: https://ibm.box.com/s/0f2w5ujd3oeb6vzfjnezdij77wb0gews

**Comment by frowe** [comment]
POC Forum review notes/comments:

Slide 2: to change "features" to "config elements"
Slide 5: validate whether openapi feature is required for /platform/validation endpoints
Slide 14: discuss how to account for recently added config attribute to use container auth on direct lookups (default w/o the option is application) and possibly update Slide 25 to include such testing
Need to do some TVT testing

**Comment by k8vance88** [comment]
> POC Forum review notes/comments:
> 
> Slide 2: to change "features" to "config elements" DONE
   Slide 5: validate whether openapi feature is required for /platform/validation endpoints. YES THIS IS TRUE. UPDATED SLIDE 21TO INDICATE 'mpOpenApi-1.0' SHOULD ALSO BE ADDED TO THE FEATURES LIST
  Slide 14: discuss how to account for recently added config attribute to use container auth on direct lookups (default w/o the option is application) and possibly update - ADDED A 'NO RESOURCE REFERENCE' TAB
Slide 25 to include such testing Need to do some TVT testing DONE

**Comment by malincoln** [comment]
discussed with Katie.  Removing "in progress" for now until work resumes.

**Comment by k8vance88** [comment]
Chile Feature Test Summary: https://github.com/OpenLiberty/open-liberty/issues/23807

**Comment by donbourne** [comment]
**Serviceability Approval Comment** - Please answer the following questions for serviceability approval:

1.  UFO -- does the UFO identify the most likely problems customers will see and identify how the feature will enable them to diagnose and solve those problems without resorting to raising a PMR?  Have these issues been addressed in the implementation?

2.  Test and Demo -- As part of the serviceability process we're asking feature teams to test and analyze common problem paths for serviceability and demo those problem paths to someone not involved in the development of the feature (eg. L2, test team, or another development team).  
a) What problem paths were tested and demonstrated? 
b) Who did you demo to?
c) Do the people you demo'd to agree that the serviceability of the demonstrated problem scenarios is sufficient to avoid PMRs for any problems customers are likely to encounter, or that L2 should be able to quickly address those problems without need to engage L3?

3.  SVT -- SVT team is often the first team to try new features and often encounters problems setting up and using them.  Note that we're not expecting SVT to do full serviceability testing -- just to sign-off on the serviceability of the problem paths they encountered.
a) Who conducted SVT tests for this feature?
b) Do they agree that the serviceability of the problems they encountered is sufficient to avoid PMRs, or that L2 should be able to quickly address those problems without need to engage L3?  

4.  Which L2 / L3 queues will handle PMRs for this feature?  Ensure they are present in the contact reference file and in the queue contact summary, and that the respective L2/L3 teams know they are supporting it.  Ask Don Bourne if you need links or more info.

5.  Does this feature add any new metrics or emit any new JSON events?  If yes, have you updated the JMX metrics reference list / Metrics reference list / JSON log events reference list in the Open Liberty docs?

**Comment by k8vance88** [comment]
@donbourne 
1. Yes, troubleshooting when the Test button does not display has been added to the UFO. There's nothing simple that we can do to detect that situation. We will document it. It's also in the blog post. All other troubleshooting should be covered by the REST apis we call. Those have been GA for awhile.
2. a) Tested that on successful test connection the user can understand easily that it was successful.
         Tested that on failure of test connection that the user was presented with information to help them understand why the test connection failed
    b) Nathan Rauh
    c) Nathan agree that it was sufficient
3. a) Brian Hanczaryk
    b) yes, SVT agrees that serviceability of any problems encountered was sufficient to avoid PMRs or that L2 should be able to quickly address those problems without engaging L3.
4. WASADM for admin center/WASCET for ds/j2c
5. No

**Comment by gnadell** [comment]
@k8vance88 
L2 has requested a live STE presentation for this feature.  The STE templates can be found at the links below.  You can use either one to create the education.
Slide Template:  https://ibm.box.com/s/1an42g7zdgmaj84w7dft0indqfgi8ffm
Github Template:  https://pages.github.ibm.com/WASL3/site/STE/about
Please upload the completed slides to the same 'STE Archive' BOX folder and let Thanh Giang (@tgiang) or Geeta Nadella (@gnadell) know they're complete or provide them the Github link.  Once the completed STE template is provided, they will provide the STE approval.  The live presentation can be scheduled with the L2 team at a later date.  Thanks!

**Comment by k8vance88** [comment]
Doc issue: https://github.com/OpenLiberty/docs/issues/6228

**Comment by aknguyen7** [comment]
> Geeta Nadella, @tngiang73 

STE https://ibm.ent.box.com/file/341395043695?s=1an42g7zdgmaj84w7dft0indqfgi8ffm

**Comment by chirp1** [comment]
Katie opened the following doc issue, which David used to approve this epic https://github.com/OpenLiberty/docs/issues/6228
